testbed/testbed/senders.go

```diff
@@ -456,3 +468,136 @@ func (pds *PrometheusDataSender) GetCollectorPort() int {
 func (pds *PrometheusDataSender) ProtocolName() string {
 	return "prometheus"
 }
+
+type FileLogWriter struct {
+	file        *os.File
+	parsersFile *os.File
+	host        string
+	port        int
+}
+
+// Ensure FileLogWriter implements MetricDataSender.
+var _ LogDataSender = (*FileLogWriter)(nil)
+
+// NewFileLogWriter creates a new data sender that will write log entries to a
+// file, to be tailed by FluentBit and sent to the collector.
+func NewFileLogWriter(host string, port int) *FileLogWriter {
+	file, err := ioutil.TempFile("", "perf-logs.json")
+	if err != nil {
+		panic("failed to create temp file")
+	}
+
+	parsersFile, err := ioutil.TempFile("", "parsers.json")
+	if err != nil {
+		panic("failed to create temp file")
+	}
+
+	f := &FileLogWriter{
+		host:        host,
+		port:        port,
+		file:        file,
+		parsersFile: parsersFile,
+	}
+	f.setupParsers()
+	return f
+}
+
+func (f *FileLogWriter) Start() error {
+	return nil
+}
+
+func (f *FileLogWriter) setupParsers() {
+	_, err := f.parsersFile.Write([]byte(`
+[PARSER]
+    Name   json
+    Format json
+    Time_Key time
+    Time_Format %d/%m/%Y:%H:%M:%S %z
+`))
+	if err != nil {
+		panic("failed to write parsers")
+	}
+
+	f.parsersFile.Close()
+}
+
+func (f *FileLogWriter) SendLogs(logs pdata.Logs) error {
+	for i := 0; i < logs.ResourceLogs().Len(); i++ {
+		for j := 0; j < logs.ResourceLogs().At(i).InstrumentationLibraryLogs().Len(); j++ {
+			ills := logs.ResourceLogs().At(i).InstrumentationLibraryLogs().At(j)
+			for k := 0; k < ills.Logs().Len(); k++ {
+				_, err := f.file.Write(append(f.convertLogToJSON(ills.Logs().At(k)), '\n'))
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func (f *FileLogWriter) convertLogToJSON(lr pdata.LogRecord) []byte {
+	rec := map[string]string{
+		"time": time.Unix(0, int64(lr.Timestamp())).Format("02/01/2006:15:04:05Z"),
+	}
+
+	if !lr.Body().IsNil() {
+		rec["log"] = lr.Body().StringVal()
+	}
+
+	lr.Attributes().ForEach(func(k string, v pdata.AttributeValue) {
+		switch v.Type() {
+		case pdata.AttributeValueSTRING:
+			rec[k] = v.StringVal()
+		case pdata.AttributeValueINT:
+			rec[k] = strconv.FormatInt(v.IntVal(), 10)
+		case pdata.AttributeValueDOUBLE:
+			rec[k] = strconv.FormatFloat(v.DoubleVal(), 'f', -1, 64)
+		case pdata.AttributeValueBOOL:
+			rec[k] = strconv.FormatBool(v.BoolVal())
+		default:
+			panic("missing case")
+		}
+	})
+	b, err := json.Marshal(rec)
+	if err != nil {
+		panic("failed to write log: " + err.Error())
+	}
+	return b
+}
+
+func (f *FileLogWriter) Flush() {
+	_ = f.file.Sync()
+}
+
+func (f *FileLogWriter) GenConfigYAMLStr() string {
```

Comment:
Since FileLogWriter is specifically for `fluentforward` should we give a more specific name, e.g. FbFileLogWriter?